### PR TITLE
ShlomiShitrit/issue30 - operator by column

### LINKED ISSUE #30: Change operators to be by column and not by "string" | "numeric"

### PR DESCRIPTION
**Pull request by ShlomiShitrit**
fix #30
This pull request refactors the structure and names of the operators in the code.

Now the operators is by column, not by string | numeric

When the developer add a custom operator, he must assign a column to it:

```const startWithBOperator = {
 name: "startsWithB",
  customFunction: function (cell: string) {
        return cell.startsWith("B");
    },
    dataType: "string",
    **column: "name"**,
    params: [
        {
            name: "cell",
            dataType: "string",
        },
    ],
}
```

The operator will be added only to this column.

for example, let's look on "name" attribute and "type" attribute in pokemon.csv:

name:
image

type:
image

Both of them are string attributes, but only "name"  have the "startWithB" operator.

### CODE REVIEW

File: DBbot/classes/app.ts [hunk 1]
```diff
@@ -29,14 +32,17 @@ export class app {
...
+        fs.writeJsonSync(tmpFilePath, bot);
+
+        process.env.DB_BOT_FILE = tmpFilePath;
 
         this.startReactApp(this.botUIPath)
             .then(() => {
                 console.log("React app started successfully");
             })
-            .catch(() => {
-                console.error("Failed to start React app");
```
Comment (nadavWeisler) [inline DBbot/classes/app.ts:39]: funny its stayed that long

**Review by nadavWeisler — APPROVED** [review]
Great work! your use of hooks is amazing

File: BotUI/src/app/components/ChatBox/ChatBox.tsx [hunk 2]
```diff
@@ -16,16 +16,17 @@ import useEndChat from "@/app/hooks/useEndChat";
...
     const [isEndSection, setIsEndSection] = useState<boolean>(false);
     const [lastQuestionIndex, setLastQuestionIndex] = useState<number>(
         botMessages.length - 1
     );
+    const [currentParameter, setCurrentParameter] = useState<string>("");
 
-    const strParam = {
-        state: isStringParameter,
-        setState: setIsStringParameter,
+    const crtParam = {
```
[the remark this answers is not in the document]
Comment (ShlomiShitrit) [reply BotUI/src/app/components/ChatBox/ChatBox.tsx:25]: change to currentParam

File: BotUI/src/app/general/interfaces.ts [hunk 3]
```diff
@@ -67,27 +67,19 @@ interface BotColumn {
...
+    operatorsArray: BotOperatorData[];
 }
-
-interface BotOperators {
-    numeric: BotOperatorData[];
-    string: BotOperatorData[];
-}
-
-interface BotOperatorData extends NameDataType {
+interface BotOperatorData {
```
[the remark this answers is not in the document]
Comment (ShlomiShitrit) [reply BotUI/src/app/general/interfaces.ts:72]: ty!

**Pull request merged**